lib/spack/spack/test/concretize_preferences.py

```diff
@@ -114,11 +114,50 @@ def test_preferred_providers(self):
         spec = concretize('mpileaks')
         assert 'zmpi' in spec
 
+    def test_preferred(self):
+        """"Test packages with some version marked as preferred=True"""
+        spec = Spec('preferred-test')
+        spec.concretize()
+        assert spec.version == spack.spec.Version('0.2.15')
+
+        # now add packages.yaml with versions other than preferred
+        conf = syaml.load("""\
+preferred-test:
+        version: [0.2.16]
+""")
+        spack.config.update_config('packages', conf, 'concretize')
```

Comment:
don't know why, but this type of changes:
```
-        conf = syaml.load("""\
 -preferred-test:
 -        version: [0.2.16]
 -""")
 -        spack.config.update_config('packages', conf, 'concretize')
 -
          # ensure that once config is in place, non-preferred version is used
 +        update_packages('preferred-test', 'version', ['0.2.16'])
```
sorted this issue. Adding `spack.package_prefs.PackagePrefs.clear_caches()` to the original code did no work.